buildIncremental.fsx

```diff
@@ -42,18 +44,37 @@ module IncrementalTests =
         | EndsWith "cmd" -> true
         | EndsWith "sh" -> true
         | _ -> false
+    
+    let getHeadHashFor repositoryDir branch =
+        let _, msg, error = runGitCommand repositoryDir (sprintf "log --oneline -1 %s" branch)
+        if error <> "" then failwithf "git log --oneline failed: %s" error
+        let logMsg = msg |> Seq.head
+        let tmp =
+            logMsg.Split(' ')
+            |> Seq.head
+            |> fun s -> s.Split('m')
+        if tmp |> Array.length > 2 then tmp.[1].Substring(0,6) else tmp.[0].Substring(0,6)
```

Comment:
This is the same as what `getCurrentHash()` does except it takes a parameter for which branch to get the HEAD for.  `git log --oneline -1 <branch>` returns something like `f90fbd84 (v1.3) Merge pull request #2857 from Aaronontheweb/v13-1.2.3-rebase` so it is just getting the first string in the stdout because that will be the hash code.